### ISSUE TITLE
Proposal: allow users to make sub-blocks glow

### ISSUE THREAD

**Issue by exKAZUu** [scratchblocks/scratchblocks#302]
Currently, `scratchblocks` allows users to make only statement-level blocks glow.
But, I believe it's useful to be able to make sub-blocks (e.g. an argument) glow.

This ticket proposes a new syntax, adding `+` immediate before a sub-block makes the sub-block glow.
e.g. `move +(1) steps`: makes only `(1)` glow

If this proposal is acceptable, then I'll make a PR. @tjvr How do you think?

**Comment by tjvr** [comment]
I think I can imagine allowing making _reporter blocks_ glow. The syntax would be something like `move (+distance to [mouse-pointer v]) steps`.

I'm not a fan of putting the `+` outside the reporter, but I can see there would be problems with putting it inside if we wanted to support string inputs, e.g. does `[+foo]` mean the string "foo" which is glowing, or the string "foo"?

I also note that a PR for this could be tricky, my recursive descent parser is probably hard to follow ;-)

**Comment by exKAZUu** [comment]
@tjvr Thank you for giving me your feedback!

>  e.g. does [+foo] mean the string "foo" which is glowing, or the string "foo"?

Sorry, could you rephrase this sentence? I'm afraid that I cannot catch your intention precisely.
Do you mean `a string/parameter "+foo"` v.s. `a glowing string/parameter "foo"`?

**Comment by tjvr** [comment]
Yeah, that's the idea.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by exKAZUu** [comment]
@tjvr Thank you for your confirmation.
Hmm, how about utilizing a space character. It is not a complete solution, but it alleviates the issue a bit.
I mean, `[+foo]` => `a string/parameter "+foo"` and `[+ foo]` => `a glowing string/parameter "foo"`
It disallows us to create `a string/parameter "+ foo"` ... How do you think?

**Comment by tjvr** [comment]
So I don't remember when I added this, but it looks like we support this already at least for variables and reporter blocks? e.g.
```
say ((foo::+ variables) + (4))
```
<img src="https://user-images.githubusercontent.com/639289/120793191-81190480-c52e-11eb-9f03-73dadb0e87ae.png" width=278>

**Comment by exKAZUu** [comment]
Thank you for the information! I think it can solve my problem. If I face the problem again, I will report again.
